pkg/kubelet/kubelet_getters.go

```diff
@@ -304,7 +311,7 @@ func (kl *Kubelet) getMountedVolumePathListFromDisk(podUID types.UID) ([]string,
 		return mountedVolumes, err
 	}
 	for _, volumePath := range volumePaths {
-		isNotMount, err := kl.mounter.IsLikelyNotMountPoint(volumePath)
+		isNotMount, err := kl.mounter.IsNotMountPoint(volumePath)
```

Comment:
I'm going to remove this commit.  Changing to IsNotMountPoint is not strictly necessary to fix this particular bug.

But we should investigate other callers of these functions to see if this may be wrong in other contexts.

Reply:
Opened up https://github.com/kubernetes/kubernetes/issues/72347 to investigate all uses of IsLikelyNotMountPoint

Reply:
SGTM

Reply:
why is it necessary to call `IsLikelyNotMountPoint` function at all then? There's no protection that it won't end up removing data if mount is from the same device anyway (which was the case for the bug we hit), so if you are ok with that, it seems like calling the function just gives false sense of confidence.

Reply:
IsLikelyNotMountPoint still works for most other volume types that are actually different devices.

The reason why I didn't want to change this function here is because it's being called in other paths as well, and we only wanted to do a targeted fix for this particular issue.  We can do a more holistic investigation in #72346 and #72347 for further areas to cleanup.